src/base/ClpMap.h

```diff
@@ -114,19 +108,26 @@ class ClpMap
     /// The number of currently stored entries, including expired ones
     size_t entries() const { return entries_.size(); }
 
-    /// read-only iterator. Entries are owned by the ClpMap, so invalidations
-    /// might happen if any non-const operation is performed on the map.
-    /// It may return expired entries. Order is not guaranteed
-    ConstEntriesIterator cbegin() { return entries_.cbegin(); }
-    ConstEntriesIterator cend() { return entries_.cend(); }
+    /// Read-only traversal of all cached entries in LRU order, least recently
+    /// used entry first. Stored expired entries (if any) are included. Any map
+    /// modification may invalidate these iterators and their derivatives.
+    ConstEntriesIterator cbegin() const { return entries_.cbegin(); }
+    ConstEntriesIterator cend() const { return entries_.cend(); }
+    ConstEntriesIterator begin() const { return cbegin(); }
+    ConstEntriesIterator end() const { return cend(); }
```

Comment:
While I see the handiness of doing this, won't it break the minimum surprise principle?

Reply:
The only surprise I can imagine here is that somebody (incorrectly) tries to modify the map using begin(), and the compiler (correctly) stops them. I think that price is worth paying for supporting range-based for loops because those loops are safer and easier to write/read. I wish C++ did not create this trade-off for us, but, unfortunately, we have to pick the lesser evil (AFAICT).

I expect future IP cache dumping code to use range-based for loops. I am not sure anybody would need to use explicit cbegin()/cend() in the foreseeable future, but I am not against adding those trivial methods, for completeness sake.

Reply:
BTW, if you want us to be explicit about it, we can do this:

```suggestion
    /// range-based `for` loop support; \sa cbegin()
    ConstEntriesIterator begin() const { return cbegin(); }
    ConstEntriesIterator end() const { return cend(); }
```